Kconfig.zephyr

```diff
@@ -149,24 +149,6 @@ config CUSTOM_LINKER_SCRIPT
 	  linker script and avoid having to change the script provided by
 	  Zephyr.
 
-config CUSTOM_RODATA_LD
-	bool "Include custom-rodata.ld"
-	help
-	  Include a customized linker script fragment for inserting additional
-	  data and linker directives into the rodata section.
-
-config CUSTOM_RWDATA_LD
-	bool "Include custom-rwdata.ld"
-	help
-	  Include a customized linker script fragment for inserting additional
-	  data and linker directives into the data section.
-
-config CUSTOM_SECTIONS_LD
-	bool "Include custom-sections.ld"
-	help
-	  Include a customized linker script fragment for inserting additional
-	  arbitrary sections.
-
```

Comment:
Before dropping support for CUSTOM_RODATA_LD we want to first mark it as deprecated to give users a reasonable window of time to migrate.

The same applies to SOC.

But we should still migrate the in-tree users.

Reply:
Is there a way to deprecate beyond mentioning it in the prompt/help text?

Reply:
Unfortunately not.

Reply:
Take a look now

Reply:
LGTM :)